be2-scala/src/main/scala/ch/epfl/pop/storage/DiskStorage.scala

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.pop.storage
+
+import java.io.{File, IOException}
+import java.nio.charset.StandardCharsets
+
+import org.iq80.leveldb.{DB, Options, WriteBatch}
+import org.iq80.leveldb.impl.Iq80DBFactory.factory
+import org.slf4j.{Logger, LoggerFactory}
+
+class DiskStorage(val databaseFolder: String = DiskStorage.DATABASE_FOLDER) {
```

Comment:
To keep the API clean, I would make sure that the API doesn't expose `org.iq80.leveldb.*` types, such as the DBException.

Otherwise the mocks will need to do that too.

It may make sense to see what API design would keep the usage simpler.
-> simple success / failure return values (or `Option`s) + internal logging ?
-> a dedicated exception that wraps the DBException ? 
-> something else?

Reply:
That's a good point! If my logic from previous message makes sense, then the wrapped exception is definitely the best in our case

Reply:
It does make sense 😄